src/modalities/checkpointing/checkpoint_saving_strategies.py

```diff
@@ -44,41 +45,41 @@ def __init__(self, k: int = -1):
                 Set to a positive integer to save the specified number of
                 checkpointsStrategy for saving the k most recent checkpoints only.
         """
-        self.saved_step_checkpoints = []
+        self.saved_step_checkpoints: List[TrainingProgress] = []
         self.k = k
 
     def get_checkpoint_instruction(
         self,
-        num_train_steps_done: int,
+        training_progress: TrainingProgress,
         evaluation_result: Dict[str, EvaluationResultBatch] | None = None,
         early_stoppping_criterion_fulfilled: bool = False,
     ) -> CheckpointingInstruction:
         """
         Generates a checkpointing instruction based on the given parameters.
 
         Args:
-            num_train_steps_done (int): The number of training steps done.
+            training_progress (TrainingProgress): The training progress.
             evaluation_result (Dict[str, EvaluationResultBatch] | None, optional):
-            The evaluation result. Defaults to None.
+                The evaluation result. Defaults to None.
             early_stoppping_criterion_fulfilled (bool, optional):
-            Whether the early stopping criterion is fulfilled. Defaults to False.
+                Whether the early stopping criterion is fulfilled. Defaults to False.
 
         Returns:
             CheckpointingInstruction: The generated checkpointing instruction.
         """
-        checkpoints_to_delete = []
+        checkpoints_to_delete: List[TrainingProgress] = []
         save_current = True
 
         if self.k > 0:
-            self.saved_step_checkpoints = [num_train_steps_done] + self.saved_step_checkpoints
+            self.saved_step_checkpoints = [training_progress] + self.saved_step_checkpoints
```

Comment:
Should that not be training_progress.num_seen_steps_total instead of training_progress?

Reply:
with the TrainingProgress (see https://github.com/Modalities/modalities/blob/6d550b618f5d80952eb18b53fc840eabf3122944/src/modalities/training/training_progress.py)
we store the current training progress (including num_seen_steps_total). 

To identify a checkpoint however, we need additional progress information such as num_seen_tokens:
```
eid_{experiment_id}-{entity}-seen_steps_{num_seen_steps}-seen_tokens_{num_seen_tokens}-target_steps_{num_target_steps}-target_tokens_{num_target_tokens}.bin
```
That's why, we pass around the `TrainingProgress`